src/Hubbup.MikLabelModel/DiffHelper.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Hubbup.MikLabelModel
+{
+    public struct SegmentedDiff
+    {
+        public string[] FileDiffs { get; set; }
+        public IEnumerable<string> Filenames { get; set; }
+        public IEnumerable<string> Extensions { get; set; }
+        public Dictionary<string, int> Folders { get; set; }
+        public Dictionary<string, int> FolderNames { get; set; }
+        public bool AddDocInfo { get; set; }
+        public bool PossiblyExtensionsLabel { get; set; }
+    }
+
+    public class DiffHelper
+    {
+        /// <summary>
+        /// name of files taken from fileDiffs
+        /// </summary>
+        public IEnumerable<string> FilenamesOf(string[] fileDiffs) => fileDiffs.Select(fileWithDiff => Path.GetFileNameWithoutExtension(fileWithDiff));
+
+        /// <summary>
+        /// file extensions taken from fileDiffs
+        /// </summary>
+        public IEnumerable<string> ExtensionsOf(string[] fileDiffs) => fileDiffs.Select(file => Path.GetExtension(file)).
+                Select(extension => string.IsNullOrEmpty(extension) ? "no_extension" : extension);
+
+        public SegmentedDiff SegmentDiff(string[] fileDiffs)
+        {
+            if (fileDiffs == null || string.IsNullOrEmpty(string.Join(';', fileDiffs)))
+            {
+                throw new ArgumentNullException(nameof(fileDiffs));
+            }
+            var folderNames = new Dictionary<string, int>();
+            var folders = new Dictionary<string, int>();
+            bool addDocInfo = false, possiblyExtensionsLabel = false;
+            string folderWithDiff, subfolder;
+            string[] folderNamesInPr;
+            foreach (var fileWithDiff in fileDiffs)
+            {
+                folderWithDiff = Path.GetDirectoryName(fileWithDiff) ?? string.Empty;
+                folderNamesInPr = folderWithDiff.Split(Path.DirectorySeparatorChar);
```

Comment:
Ah, so `Path.GetDirectoryName()` appears to normalize file paths to Windows-style paths. I don't think this is a safe assumption to make for x-plat, and it's also a very unexpected behavior of this API (even if it is documented as such). I would recommend passing in both directory separator chars by using the overload `string[] Split (params char[] separator);` as in:

```c#
                folderNamesInPr = folderWithDiff.Split(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
```

I think this will make the code both cleaner and more robust.